hawc_hal/HAL.py

```diff
@@ -166,8 +166,12 @@ def _setup_psf_convolutors(self):
 
         self._psf_convolutors = collections.OrderedDict()
         for bin_id in central_response_bins:
-            self._psf_convolutors[bin_id] = PSFConvolutor(central_response_bins[bin_id].psf, self._flat_sky_projection)
+        
+            #Only set up PSF convolutors for active bins.
+            if bin_id in self._active_planes:
```

Comment:
It'd crash. I added a  fix for that  (calling set_model after changing active  planes). I don't think it should add too much processing  time since presumably  the used would not change the active planes too often).